### PR TITLE
initial support motion photos

### PR DESCRIPTION
**Pull request by sickelap**
This adds support for parsing and extracting video from Google Pixel and Samsung motion photos.
Sample photo attached to https://github.com/LibrePhotos/librephotos/issues/797 did not have XMP metadata and exiftool was unable to parse it. Parsing manually does pretty good job. Extracted videos are added as `embedded_media` property from `Photo.main_file`.

Adding support for `HEIF`/`HEIC` (a.k.a. Apple Live Photos) is more involved, but hopefully we will have it done as well in the future.

PR for frontend will follow

### CODE REVIEW

File: api/migrations/0042_add_embedded_media.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("api", "0041_apply_user_enum_for_person"),
```
Comment (derneuere) [inline api/migrations/0042_add_embedded_media.py]: We now have a couple of other migrations, which means, that this line needs to be updated

**Comment by derneuere** [comment]
I would prefer it if we could maintain consistency in how we save technical files for LibrePhotos.  "embedded_media" files are extracted in the folder where the original file lives. This breaks the regular pattern where we save needed files in the protected_media folder. I think adding a folder named embedded_media and saving embedded_media there would make the most sense.

I think there are also a couple of changes needed for calculating the thumbnails. The thumbnail in the list is probably a video and the big thumbnail an image. We also would need a new type of live / motion photos, which you could then add to PigSerializer.

I am looking forward to your frontend pull request!

**Comment by sickelap** [comment]
I totally agree with embedded media location. As those are not primary files but rather extracted from the original, it makes sense to keep them in a private location. In fact, this was bugging me too. Now, extracted media files will be placed into the `/protected_media/embedded_media/`.

I don't think that messing with thumbnails is necessary because of what I have in mind for the frontend.
The idea I am working towards is not to display embedded media as standalone content. Instead, when opening the motion photo in light box, we will have a “Play” button (similar to how it's done in google photos). I also don't think we need to have a dedicated type for embedded media and the extracted content is a M2M relationship to the `Photo.main_file`. This makes implementation simpler (I think) and not as intrusive as it would have been otherwise.

**Comment by derneuere** [comment]
Alright, looking forward to the frontend pr :)

**Pull request merged**